app/src/main/java/com/nutomic/syncthingandroid/activities/DeviceActivity.java

```diff
@@ -214,7 +214,7 @@ private void restoreDialogStates(Bundle savedInstanceState) {
     public void onDestroy() {
         super.onDestroy();
         if (getService() != null) {
-            getService().unregisterOnApiChangeListener(this::onApiChange);
+            getService().unregisterOnServiceStateChangeListener(this::onServiceStateChange);
```

Comment:
What if getServics returns a null on the second invocation?

Reply:
improved with commit https://github.com/Catfriend1/syncthing-android/commit/66ed4bdda02820267bd1dc33b31561a8eac1b54e
This is just a safety check already present on the old code. But yes, I agree storing the ref on a variable and use this is better so I did improve that. As we shutdown the syncthing service first an d then finish MainActivity (holding this fragment) which calls onDestroy() getService couldn't get null while this code part is running at the moment (but in a future implementation maybe :) )